File: Cristina - 100711128/kafka-ptyhon/consumer.py
import json
from kafka import KafkaConsumer

if __name__ == '__main__':
    # Kafka Consumer
    consumer = KafkaConsumer(
        'topic-testpy',
        bootstrap_servers='localhost:9094',
        auto_offset_reset='earliest'
    )
    print("Connected!")

    # Print messages
    for message in consumer:
        print(json.loads(message.value))

File: Austin Page - 100725236/Nodejs code/consumer.js
const { Kafka } = require("kafkajs");

run();

async function run() {
  try {
    const kafka = new Kafka({
      brokers: ["localhost:9093"],
    });

    const consumer = kafka.consumer({ groupId: "test" });
    console.log("Connecting...");
    await consumer.connect();
    console.log("Connected!");

    consumer.subscribe({
      topic: "test",
      fromBeginning: true,
    });

    await consumer.run({
      eachMessage: async (result) => {
        console.log(
          "Received: " +
            result.message.value +
            " on partition: " +
            result.partition
        );
      },
    });
  } catch (err) {
    console.error("Error: " + err);
  }
}
